src/ImageSharp/Processing/Transforms/TransformHelpers.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) Six Labors and contributors.
+// Licensed under the Apache License, Version 2.0.
+
+using System;
+using System.Numerics;
+using SixLabors.Primitives;
+
+namespace SixLabors.ImageSharp
+{
+    /// <summary>
+    /// Contains helper methods for working with affine transforms
+    /// </summary>
+    public class TransformHelpers
+    {
+        /// <summary>
+        /// Returns the bounding <see cref="Rectangle"/> relative to the source for the given transformation matrix.
+        /// </summary>
+        /// <param name="rectangle">The source rectangle.</param>
+        /// <param name="matrix">The transformation matrix.</param>
+        /// <returns>
+        /// The <see cref="Rectangle"/>.
+        /// </returns>
+        public static Rectangle GetTransformedBoundingRectangle(Rectangle rectangle, Matrix3x2 matrix)
```

Comment:
This method needs a review + extensive unit testing including cases having a rectangle with a negative origo.
I suggest to internalize it in this PR instead, so we can focus on the rest.

Reply:
OK, will do.